addons/payment_razorpay/tests/test_payment_transaction.py

```diff
@@ -50,6 +50,27 @@ def test_void_is_not_supported(self):
         tx = self._create_transaction('direct', state='authorized')
         self.assertRaises(UserError, func=tx._send_void_request)
 
+    def test_prevent_multi_payments_on_recurring_transactions(self):
+        """ Test that no retry is allowed within the 36 hours of the first attempt using token. """
+        shared_token = self._create_token()
+        tx1 = self._create_transaction('token', reference='INV123', token_id=shared_token.id)
+        tx1.state = 'pending'
+        tx2 = self._create_transaction('token', reference='INV123-1', token_id=shared_token.id)
+        self.assertRaises(UserError, tx2._send_payment_request)
+
+    def test_allow_recurring_transactions(self):
+        """ Test different transactions with same token can still be created. """
+        shared_token = self._create_token(provider_ref='cust_123,token_404')
+        tx1 = self._create_transaction('token',reference='INV123', token_id=shared_token.id)
+        tx1.state = 'pending'
+        tx2 = self._create_transaction('token', reference='INV456', token_id=shared_token.id)
```

Comment:
In the same test, you could loop on a set of transactions with a different setup each time: different ref, different token, not pending, etc.